Opcode 8XYK

ilnoumri opened the pull request:
#17 #18 #19 #21 #23 #25 #27 #30 #32

Commit:
Adding 8XY(0 to E)

Commit:
Adding 8XY(0 to E)

alarsyo commented:
Je regarde celle ci dès que j'ai un PC sous la main, elle est un peu longue sinon xD

megastorm7 commented on Chip8/Chip8/OpCodes.fs:
```diff
@@ -30,6 +30,65 @@ let instruction_7XKK opcode = let X = int (opcode &&& 0x0F00us >>> 8) in //garde
+                              chip8.PC <- chip8.PC + 2us
+
+//8XY5 Affecter VX-VY à VX.Si VX > VY alors VX[F] = 1 sinon VX[F] = 0
+let instruction_8XY5 opcode = let X = int (opcode &&& 0x0F00us >>> 8) in
```
/!\

alarsyo commented on Chip8/Chip8/OpCodes.fs:
```diff
@@ -30,6 +30,65 @@ let instruction_7XKK opcode = let X = int (opcode &&& 0x0F00us >>> 8) in //garde
+let instruction_8XY4 opcode = let X = int (opcode &&& 0x0F00us >>> 8) in
+                              let Y = int (opcode &&& 0x00F0us >>> 4) in
+                              chip8.Vx.[X] <- chip8.Vx.[X] + chip8.Vx.[Y]
+                              if (uint16 X + uint16 Y > 0xFFus) then chip8.Vx.[0xF] <- 1uy else chip8.Vx.[0xF] <- 0uy
```
à changer, t'additionne les indices là.... x))

alarsyo commented on Chip8/Chip8/OpCodes.fs:
```diff
@@ -30,6 +30,65 @@ let instruction_7XKK opcode = let X = int (opcode &&& 0x0F00us >>> 8) in //garde
+let instruction_8XY5 opcode = let X = int (opcode &&& 0x0F00us >>> 8) in
+                              let Y = int (opcode &&& 0x00F0us >>> 4) in
+                              chip8.Vx.[X] <- chip8.Vx.[X] - chip8.Vx.[Y] 
+                              if (chip8.Vx.[X] > chip8.Vx.[Y]) then chip8.Vx.[0xF] <- 1uy else chip8.Vx.[0xF] <- 0uy
```
ta comparaison tu la fais après avoir assigné une nouvelle valeur à `Vx.[X]`

alarsyo commented on Chip8/Chip8/OpCodes.fs:
```diff
@@ -30,6 +30,65 @@ let instruction_7XKK opcode = let X = int (opcode &&& 0x0F00us >>> 8) in //garde
+let instruction_8XY6 opcode = let X = int (opcode &&& 0x0F00us >>> 8) in
+                              let Y = int (opcode &&& 0x00F0us >>> 4) in
+                              chip8.Vx.[0xF] <- chip8.Vx.[X] &&& 0b0000001uy
+                              chip8.Vx.[X] <- chip8.Vx.[Y] >>> 1
```
`Vx.[Y]` est pas utilisé là

alarsyo commented on Chip8/Chip8/OpCodes.fs:
```diff
@@ -30,6 +30,65 @@ let instruction_7XKK opcode = let X = int (opcode &&& 0x0F00us >>> 8) in //garde
+                              chip8.PC <- chip8.PC + 2us
+
+//8XY7 Affecter VX-VY à VX.Si VX < VY alors VX[F] = 1 sinon VX[F] = 0
+let instruction_8XY7 opcode = let X = int (opcode &&& 0x0F00us >>> 8) in
```
Celui là c'est un copié collé du 8XY5, t'as pas modifié ce qu'il fallait

Commit:
8XY4,8XY5 and 8XY7 modification

Commit:
8XY4 real modification

Commit:
8XY4 modification

alarsyo committed:
Symbolic commit to close issues automagically

Close #17
Close #18
Close #19
Close #21
Close #23
Close #25
Close #27
Close #30
Close #32

Signed-off-by: Antoine Martin <antoine97.martin@gmail.com>

RitoFan reviewed (commented):
짜증

Commit:
Fixing 8XYE

Commit:
Fixing 8XYE

Commit:
Removing useless Y values

alarsyo reviewed (approved):
Utsukushii

alarsyo committed:
Merge branch 'master' into opcode/8XYK
